enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,964 @@
+
+  // IP address on VirtualMachine's network interfaces included in the fields' CIDRs
+  // that will be used in respective status.addresses fields.
+  // +kubebuilder:validation:Optional
+  NetworkSubnetCIDR string `json:"networkSubnetCidr,omitempty"`
+
+  // IP address on VirtualMachine's VM Network names that will be used to when searching
+  // for status.addresses fields. Note that if internal.networkSubnetCidr and
+  // external.networkSubnetCidr are not set, then the vNIC associated to this network must
+  // only have a single IP address assigned to it.
+  // +kubebuilder:validation:Optional
+  Network string `json:"network,omitempty"`
+
+  // IP addresses in these subnet ranges will be excluded when selecting
+  // the IP address from the VirtualMachine's VM for use in the
+  // status.addresses fields.
+  // +kubebuilder:validation:Optional
+  ExcludeNetworkSubnetCIDR string `json:"excludeNetworkSubnetCidr,omitempty"`
+}
+
+// VSpherePlatformNodeNewtorking
+//
+type VSpherePlatformNodeNetworking struct {
+
+  // external is the IP selection configuration for
+  // a node's external node address.
+  // +kubebuilder:validation:Optional
+  External VSpherePlatformNodeNetworkingSpec `json:"external,omitempty"`
+
+  // internal is the IP selection configuration for
+  // a node's internal node address.
+  // +kubebuilder:validation:Optional
+  Internal VSpherePlatformNodeNetworkingSpec `json:"internal,omitempty"`
+}
+
+// VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type VSpherePlatformSpec struct {
+  // vcenters holds the connection details for services to communicate with vCenter.
+  // Currently only a single vCenter is supported.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MaxItems=1
+  // +kubebuilder:validation:MinItems=1
+  VCenters []VSpherePlatformVCenterSpec `json:"vcenters,omitempty"`
+
+  // deploymentZones holds the association between vcenter, failure domains
+  // and vcenter placement for virtual machines.
+  // +kubebuilder:validation:Optional
+  DeploymentZones []VSpherePlatformDeploymentZoneSpec `json:"deploymentZones,omitempty"`
+
+  // failureDomains holds the VSpherePlatformFailureDomainSpec which contains
+  // the definition of region, zone and the vCenter topology.
+  // If this is omitted failure domains (regions and zones) will not be used.
+  // +kubebuilder:validation:Optional
+  FailureDomains []VSpherePlatformFailureDomainSpec `json:"failureDomains,omitempty"`
+
+  // nodeNetworking defines the way that the Node IPs are selected
+  // from the addresses assigned to the Node in kube-api.
+  // If nodeNetwork is not provided, default selection
+  // will select the first address that is not a Localhost address.
+  // +kubebuilder:validation:Optional
+  NodeNetworking VSpherePlatformNodeNetworking `json:"nodeNetworking,omitempty"`
+}
+```
+
+### CCCMO
```

Comment:
ccmo is external cloud controller operator?

Reply:
Cluster Cloud Controller Manager Operator
https://github.com/openshift/cluster-cloud-controller-manager-operator